flask_resty/jwt.py

```diff
@@ -89,3 +96,59 @@ def get_config_key(self, key):
 
     def get_credentials(self, payload):
         return payload
+
+
+class JwkSetAuthentication(JwtAuthentication):
+    def __init__(self, jwk_set=None, **kwargs):
+        super(JwkSetAuthentication, self).__init__(**kwargs)
+
+        self.jwks = jwk_set
+        self.algorithms = get_default_algorithms()
+
+    def get_jwks(self):
+        config = flask.current_app.config
+        return self.jwks \
+            if self.jwks else config[self.get_config_key('jwk_set')]
+
+    def key_from_jwk(self, jwk, algorithm):
+        if 'x5c' in jwk:
+            return load_der_x509_certificate(
+                base64.b64decode(jwk['x5c'][0]),
+                default_backend(),
+            ).public_key()
+
+        # awkward
+        return algorithm.from_jwk(json.dumps(jwk))
+
+    def get_jwk_for_token(self, token):
+        unverified_header = jwt.get_unverified_header(token)
+
+        try:
+            token_kid = unverified_header['kid']
+        except KeyError:
+            raise InvalidTokenError('Key ID header parameter is missing')
+
+        for jwk in self.get_jwks()['keys']:
+            if jwk['kid'] == token_kid:
+                return jwk
+
+        raise InvalidTokenError("no key found")
+
+    def decode_token(self, token):
+        args = self.get_jwt_decode_args()
+
+        unverified_header = jwt.get_unverified_header(token)
+        jwk = self.get_jwk_for_token(token)
+
+        alg = jwk['alg'] if 'alg' in jwk else unverified_header['alg']
+
+        # I'm not sure we need to re-up this check here, jwt.decoode will
+        # check and raise the the same thing
+        if alg not in args['algorithms']:
+            raise InvalidAlgorithmError(
```

Comment:
Would you mind getting a coverage hit on here? Just try a token with `"alg": "none"`. It's a good check to have.